src/components/button/button.svelte

```diff
@@ -79,7 +80,22 @@
   on:click={onClick}
   {...$$restProps}
 >
-  <slot>Leo Button</slot>
+  {#if isLoading}
+    {#if $$slots.loading}
+      <slot name="loading" />
+    {:else if !fab}
+      <div>
+        <slot>Leo Button</slot>
+      </div>
+    {/if}
+    <ProgressRing />
+  {:else}
+    <slot name="left-icon" />
+    <div>
```

Comment:
I believe so, unfortunately. Since we're using flexbox to manage the spacing, and we don't know what elements are going to be passed into the `<slot/>`, we need to ensure only three immediate descendants of the `<button>` (or `<a>`) tag.

I suppose we could just apply a margin on each icon instead of using `flex` for the parent, though...

Reply:
Ah right :D I always forget that slotted content can be multiple items :laughing: Okay, I'll leave this up to you :)

Reply:
I don't like having extra markup, however given that the slotted contents can be literally anything, I think wrapping it in a div gives us some guarantees about icon layout that we can't otherwise have.